### PR TITLE
feat(action): move preflight steps to custom deploy action

### PR DESCRIPTION
**Pull request by IamAbbey**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -79,72 +83,20 @@ runs:
         aws_secret_access_key: ${{ inputs.aws_secret_access_key }}
         aws_region: ${{ inputs.aws_region }}
 
-    - name: Render and deploy preflight task definition to Amazon ECS
-      id: deploy-preflight-task-definition
-      uses: moneymeets/action-ecs-deploy/custom-deploy-steps/create-task-definition@master
+    - name: Deploy task definition and run preflight
+      id: deploy-definition-and-run-preflight
```
Comment (felix11h) [inline action.yml]: Probably _task_ definition missing in the name here. How about
```suggestion
      id: deploy-preflight-task-definition-and-run-task
```
?

**Review by felix11h — COMMENTED** [review]
Just a change in the naming of the step, looks good to me otherwise!

File: action.yml [hunk 2]
```diff
@@ -14,6 +14,10 @@ inputs:
     description: Number of tasks running in ECS service
     required: false
     default: 2
+  run_preflight:
+    description: Determines whether the preflight step should be triggered
+    required: false
+    default: "true"
```
Comment (marns93) [inline action.yml]:
```suggestion
    default: true
```

File: custom-deploy-steps/run-preflight/action.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
+name: Get image URI
+description: Get image uri from AWS ECR
```
Comment (marns93) [inline custom-deploy-steps/run-preflight/action.yml]: Please update

File: custom-deploy-steps/run-preflight/action.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  aws_secret_access_key:
+    description: AWS secret access key
+    required: true
+  aws_region:
+    description: AWS region
+    required: true
+
+outputs:
+  preflight-task-arn:
+    description: "Preflight task ARN"
```
Comment (marns93) [inline custom-deploy-steps/run-preflight/action.yml]:
```suggestion
    description: Preflight task ARN
```

File: custom-deploy-steps/run-preflight/action.yml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    value: ${{ steps.run-preflight.outputs.preflight-task-arn }}
+  previous-task-definition-arn:
+    description: "ARN of previous task definition revision"
```
Comment (marns93) [inline custom-deploy-steps/run-preflight/action.yml]:
```suggestion
    description: ARN of previous task definition revision
```

File: custom-deploy-steps/run-preflight/action.yml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    value: ${{ steps.deploy-preflight-task-definition.outputs.previous-task-definition-arn }}
+  latest-task-definition-arn:
+    description: "ARN of new task definition revision"
```
Comment (marns93) [inline custom-deploy-steps/run-preflight/action.yml]:
```suggestion
    description: ARN of new task definition revision
```

**Review by marns93 — CHANGES_REQUESTED** [review]
In general it looks good to me. Just smaller things. We agreed on removing quotes if they are not needed.
Did you test the changes? Perhaps you can link the pipelines in this PR?

**Comment by IamAbbey** [comment]
Test Action Runs
with preflight=true
https://github.com/moneymeets/django-insurance-api-v2/actions/runs/10416562236/workflow 

with preflight=false
https://github.com/moneymeets/django-insurance-api-v2/actions/runs/10416929798/workflow

cc @marns93

**Review by marns93 — APPROVED** [review]
:+1:

**Pull request merged**